### PR TITLE
feat(dreamzero): add DreamZero LIBERO 14B SFT (World-Action Model) test case on EKS

### PR DESCRIPTION
**Pull request by bluecrayon52**
## What

A self-contained test case for continue-SFT (supervised fine-tuning) of the **DreamZero** 14B World-Action Model (WAM) on the **LIBERO** manipulation benchmark, on Amazon EKS. It demonstrates multi-node **FSDP2 + KubeRay RayJob** training over **EFA**, with sharded DCP (Distributed Checkpoint) saving, an offline DCP→`.pt` conversion, and in-simulator LIBERO eval.

Everything lives under `3.test_cases/pytorch/dreamzero/` (24 files, +2,762 lines, no changes elsewhere). The image is a two-stage build: the upstream RLinf `embodied-libero` target + an EFA/NCCL overlay.

## Hardware

- Training: **2× `p5en.48xlarge`** (8× H200 each = 16 GPUs), 16 EFA NICs/node
- Eval: 1× `p5en.48xlarge`
- Shared storage: FSx for Lustre (≥250 GB free; a 14B DCP checkpoint is ~140–206 GB)

## Validation status

- **Build (L1/L2):** builds via `docker buildx` (two-stage) and was validated end-to-end through CodeBuild + a 10/10 container test on `p5en.48xlarge` (CUDA, EFA device + libs, multi-venv, RLinf import, patch-applied invariants, FSx, Ray, OpenMPI).
- **Training:** a real **300-step** SFT run on 2× `p5en.48xlarge` reduced `train/loss` **0.232 → 0.085** (~6.9 s/step) and wrote a **207 GB** sharded DCP checkpoint with no corruption or save-time crashes. This demonstrates the pipeline *trains and converges* — it is **not** a task-accuracy claim (300 steps is short; the released 14B trained for 100K).
- **Eval:** the LIBERO simulator eval runs and renders an in-sim rollout video; `eval/success_once = 0.0` is expected for a short run.

## Notes for reviewers

- **Non-commercial model license:** the `GEAR-Dreams/DreamZero-DROID` checkpoint is released under **CC-BY-NC-4.0** (non-commercial). This is called out prominently at the top of both READMEs; production use needs additional approvals. The repository code remains **MIT-0**.
- **Git LFS required** to view the `.svg` diagrams (`diagrams/*.drawio.svg` are LFS pointers).
- **Fully parameterized** — `envsubst` + env vars; no file editing needed to run the walkthrough. No secrets committed (only `*.example` files).
- **Upstream pins:** RLinf `b3bbabb1f461`, dreamzero/`groot` `ab790c198fbc`, and the `GEAR-Dreams/DreamZero-DROID` checkpoint.
- Ships `dcp-save-gloo-coordinator.patch` — a DCP-finalization fix (passes a gloo process group to `dcp.save` so the post-write object-broadcast avoids the NCCL teardown race). Verified necessary regardless of torch version (the synchronous `dcp.save` path is unchanged through ≥ torch 2.8).

## Layout

- `Dockerfile` + `dcp-save-gloo-coordinator.patch` — two-stage RLinf + EFA overlay image
- `kubernetes/libero/` — the EKS recipe: `build-push.sh`, `model-download.yaml`, `generate-metadata.yaml`, `dreamzero-sft.yaml` (RayJob), `convert-checkpoint.yaml`, `dreamzero-eval.yaml`, launcher scripts, optional FSx storage + HF-token templates, and a full step-by-step `README.md`
- `diagrams/` — WAM architecture + multi-node infra topology

### CODE REVIEW

File: 3.test_cases/pytorch/dreamzero/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Transient build clones created by kubernetes/libero/setup/build-push.sh
```
Comment (mvinci12) [inline 3.test_cases/pytorch/dreamzero/.gitignore]: Stale path in the comment: this references `kubernetes/libero/setup/build-push.sh`, but the script was moved out of `setup/` to `kubernetes/libero/build-push.sh` (commit 58477a8). The ignore rules below (`/RLinf/`, `/DreamZero/`) are correct — just update the comment to drop `setup/`.

File: 3.test_cases/pytorch/dreamzero/kubernetes/libero/build-push.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+set -euo pipefail
+
+: "${ECR_URI:?set ECR_URI (e.g. <acct>.dkr.ecr.<region>.amazonaws.com/dreamzero)}"
+: "${AWS_REGION:?set AWS_REGION}"
+UPSTREAM_REPO="${UPSTREAM_REPO:-https://github.com/RLinf/RLinf.git}"
+UPSTREAM_REF="${UPSTREAM_REF:-b3bbabb1f461}"
+DREAMZERO_REPO="${DREAMZERO_REPO:-https://github.com/RLinf/dreamzero.git}"
+DREAMZERO_REF="${DREAMZERO_REF:-ab790c198fbc}"
+BUILD_TARGET="${BUILD_TARGET:-embodied-libero}"
+TAG="${TAG:-latest}"
```
Comment (mvinci12) [inline 3.test_cases/pytorch/dreamzero/kubernetes/libero/build-push.sh]: `TAG` defaults to `latest`, and stage 2 pushes `${ECR_URI}:${TAG}` while every manifest pins `image: ${ECR_URI}:latest`. A mutable `:latest` tag makes the image non-reproducible and is the same issue CONTRIBUTING calls out ("do not use a latest tag") — it was part of why the kaniko path was dropped (commit dbb6404). Recommend defaulting to an explicit/immutable version tag (e.g. derived from `UPSTREAM_REF`/`DREAMZERO_REF` or a date/semver), and updating the `${ECR_URI}:latest` references in the manifests (or documenting a single `IMAGE_TAG` env that flows into both `build-push.sh` and the rendered YAML) so the pushed image and the deployed image are guaranteed to match.

File: 3.test_cases/pytorch/dreamzero/kubernetes/libero/convert-checkpoint.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              cpu: "8"
+              memory: "64Gi"
+            limits:
+              cpu: "16"
+              memory: "128Gi"
+          volumeMounts:
+            - name: fsx
+              mountPath: /fsx
+            - name: scripts
+              mountPath: /opt/scripts
```
Comment (mvinci12) [inline 3.test_cases/pytorch/dreamzero/kubernetes/libero/convert-checkpoint.yaml:70]: Launcher mount path is inconsistent across the manifests: SFT (`dreamzero-sft.yaml`) mounts its launcher ConfigMap at `/tmp/scripts`, while convert and eval use `/opt/scripts`. It works, but standardizing on one path (and matching `command:` accordingly) would make the recipe easier to follow and less error-prone when copy-adapting between steps.

**Review by mvinci12 — CHANGES_REQUESTED** [review]
Thanks for an exceptionally thorough and well-documented contribution — this is one of the more carefully validated test cases I've reviewed.

## Strengths

- **Genuinely validated end-to-end.** A real 300-step SFT run on 2× `p5en.48xlarge` (16× H200, FSDP2 `full_shard` over EFA) reducing `train/loss` 0.232 → 0.085, a 207 GB sharded DCP checkpoint written cleanly, converted to `.pt`, and consumed by the LIBERO sim eval. The KubeRay RayJob reaching `SUCCEEDED` is shown, not asserted.
- **Honest framing.** Explicitly *not* a task-accuracy claim — `eval/success_once = 0.0` for a 1-step checkpoint is documented as expected, with the upstream 5B ~96.7%@18k reference provided as the convergence baseline. No fabricated numbers.
- **Real root-cause engineering.** The `dcp-save-gloo-coordinator.patch` (gloo PG for the `dcp.save` finalization broadcast to dodge the NCCL-teardown race) is correctly diagnosed and the fix is sound; the note that it is *not* torch-version-specific through ≥2.8 is appreciated.
- **Clean security posture.** No secrets committed (`*.example` files, `secretKeyRef`, gitignored `env_vars`), prominent CC-BY-NC-4.0 model-license notice at the top of both READMEs, code remains MIT-0.
- **Consistent with repo conventions.** Layout mirrors the sibling `openvla-oft/kubernetes/libero/`; SPDX headers present; the restricted-`envsubst` allow-list pattern is correctly documented to protect inline shell vars; `set -euo pipefail` + `PYTHONPATH` unbound-guard handled correctly throughout.
- Excellent troubleshooting table and configuration deep-dive — the `num_action_per_block`, Hydra `+` prefix, and `save_full_model_weights=false` rationales will save the next person hours.

## Requesting changes

Three small items before merge (all inline). None are correctness blockers for the validated path, but #2 affects reproducibility and aligns with CONTRIBUTING:

1. `.gitignore` — comment references a stale `setup/` path.
2. `build-push.sh` — pushes `:latest`; recommend an explicit version tag.
3. `convert-checkpoint.yaml` — launcher mount path inconsistent with the other manifests.

Details inline. Thanks again — happy to re-review promptly.

File: 3.test_cases/pytorch/dreamzero/.gitignore (hunk 1 above)
Comment (bluecrayon52) [reply 3.test_cases/pytorch/dreamzero/.gitignore]: Fixed in 802bdf5 — dropped `setup/` from the comment so it points at `kubernetes/libero/build-push.sh`. Ignore rules left as-is.

File: 3.test_cases/pytorch/dreamzero/kubernetes/libero/build-push.sh (hunk 2 above)
Comment (bluecrayon52) [reply 3.test_cases/pytorch/dreamzero/kubernetes/libero/build-push.sh]: Fixed in 3262f60 by going the single `IMAGE_TAG` route. `build-push.sh` now defaults `IMAGE_TAG` to an immutable `dz-${DREAMZERO_REF}` and pushes `${ECR_URI}:${IMAGE_TAG}`. The same `IMAGE_TAG` is threaded into every manifest's `image:` reference and added to each restricted `envsubst` allow-list (so `${IMAGE_TAG}` gets substituted alongside `${ECR_URI}`/`${NAMESPACE}`), and it's documented in `env_vars.example` + the README env-var table. No more `:latest` anywhere — pushed and deployed images are now guaranteed to match.

File: 3.test_cases/pytorch/dreamzero/kubernetes/libero/convert-checkpoint.yaml (hunk 3 above)
Comment (bluecrayon52) [reply 3.test_cases/pytorch/dreamzero/kubernetes/libero/convert-checkpoint.yaml:70]: Fixed in b14e0ad — standardized on `/opt/scripts` (the path convert and eval already use, so SFT was the odd one out). Updated the SFT launcher `mountPath`, both `entrypoint` references, and the accompanying comment to match.

**Review by mvinci12 — APPROVED** [review]
LGTM!
